twirp/client.py

```diff
@@ -22,8 +23,16 @@ def _make_request(self, *args, url, ctx, request, response_obj, **kwargs):
                 response = response_obj()
                 response.ParseFromString(resp.content)
                 return response
-            else:
+            try:
                 raise exceptions.TwirpServerException.from_json(resp.json())
+            except json.JSONDecodeError:
+                if resp.status_code == 503:
+                    code = errors.Errors.Unavailable
+                else:
+                    code = errors.Errors.Unknown
+                raise exceptions.TwirpServerException(
+                    code=code, message=resp.text
```

Comment:
Would it be worth adding the original status code and headers to the `meta` dict here? And is it ok to assume that `resp.text` can be sent as `message`?

Reply:
For context: the problem I ran into was that an intermediate load balancer was returning a 503 response with an HTML body, but all we were seeing was the JSONDecodeError. We were able to diagnose the actual source of the error only after surfacing the status, headers, and body of the actual response.

Reply:
Maybe we should send standard text `message` and `meta.status_code` and `meta.message`

https://twitchtv.github.io/twirp/docs/errors.html#http-errors-from-intermediary-proxies

Check this out!

Reply:
More generally, we should try to follow the go package as closely as possible.

https://github.com/twitchtv/twirp/blob/main/protoc-gen-twirp/generator.go#L635-L664

Reply:
Totally agree, I can add to this PR or would that warrant a new one

Reply:
You can add to this one 🙏

Reply:
@ofpiyush done!